lib/src/command/login.dart

```diff
@@ -21,16 +23,47 @@ class LoginCommand extends PubCommand {
   @override
   String get invocation => 'pub login';
 
-  LoginCommand();
+  String get token => argResults['token'];
+  bool get tokenStdin => argResults['token-stdin'];
+
+  LoginCommand() {
+    argParser.addOption('token', help: 'Authorization token for the server');
+
+    argParser.addFlag('token-stdin',
+        help: 'Read authorization token from stdin stream');
```

Comment:
Didn't get that part actually? Should we prompt for token when `pub get`'ing from unauthenticated repository?

I added `--token-stdin` flag to make sure it could be used for piping credentials to pub without exposing to terminal emulator. I think it's mostly useful for CI/CD systems.

Check out: https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin

Reply:
> Should we prompt for token when pub get'ing from unauthenticated repository?

I think that's a thing we can consider supporting in the future.

------------------

But in general, when I want to sign-on my development machine. Isn't it better than I use:
*  (A) `dart pub login <server>` rather than, using
*  (B) `dart pub login <server> --token <secret>`.

Even if we are on my machine, when I do (B) the `<secret>` is liable to end up in my `~/.bash_history`, which is unfortunate.

If someday my laptop is compromised by malicious code the first thing it would reasonably steal is `~/.bash_history`.

So I'm suggesting we only support (A), and never implement (B). We simply require that the token is always entered over stdin. If a user really doesn't want to do that, then the user can do `echo <secret>  | dart pub login <server>`.

Reply:
Hey @jonasfj,

Fixed ✅

- `pub login` - Login to pub.dev
- `pub login --server <server>` - Login to given server